fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -373,6 +373,16 @@ private void ruleBasedMaterializedViewRewrite(OptExpression tree,
             // now add single table materialized view rewrite rules in rule based rewrite phase to boost optimization
             ruleRewriteIterative(tree, rootTaskContext, RuleSetType.SINGLE_TABLE_MV_REWRITE);
         }
+
+        if (MvUtils.isAppliedMVUnionRewrite(tree)) {
```

Comment:
This is used to push down filter above union down. Only support it in RBO.

CBO will use this optimization but is still a valid plan. we can do it later if needed.
```
    // NOTE: Since union rewrite will generate Filter -> Union -> OlapScan -> OlapScan, need to push filter below Union
        // and do partition predicate again.
        // TODO: Do it in CBO if needed later.
```